### PR TITLE
Enable s390x image cross platform build for calico

### PR DESCRIPTION
**Pull request by huoqifeng**
## Description

After get the `go-build` s390x image via https://github.com/projectcalico/go-build/pull/379 and all Dockerfile.s390x work well via https://github.com/projectcalico/calico/pull/5793. Will enable the build in CI.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: apiserver/Makefile [hunk 1]
```diff
@@ -124,7 +124,11 @@ sub-image-%:
 
 $(API_SERVER_IMAGE): register $(CONTAINER_CREATED)
 $(CONTAINER_CREATED): docker-image/Dockerfile.$(ARCH) $(BINDIR)/apiserver-$(ARCH) $(BINDIR)/filecheck-$(ARCH)
+ifeq ($(ARCH),s390x)
+	$(DOCKER_BUILD) -t $(API_SERVER_IMAGE):latest-$(ARCH) -f docker-image/Dockerfile.$(ARCH) .
```
[the remark this answers is not in the document]
Comment (huoqifeng) [reply apiserver/Makefile:128]: @caseydavenport thanks for the kind review. There is no s390x images available in dockerhub, usually, user need build the images on a IBM Z (s390x) machine and then run the scripts. `-- pull` and `--load` option does not work for this case.

[the remark this answers is not in the document]
Comment (caseydavenport) [reply apiserver/Makefile:128]: I'm not sure I understand - without `--load` the s390x image won't be built as part of our release process.

Do you mean that even on `s390x` machines, the `--pull` option doesn't exist in docker?

[the remark this answers is not in the document]
Comment (huoqifeng) [reply apiserver/Makefile:128]: @caseydavenport yes, I think there is `--load` option in s390x docker, here is what I see:
```
# docker version
Client:
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.13.8
 Git commit:        20.10.7-0ubuntu5~18.04.3
 Built:             Mon Nov  1 01:04:02 2021
 OS/Arch:           linux/s390x
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       20.10.7-0ubuntu5~18.04.3
  Built:            Fri Oct 22 00:57:37 2021
  OS/Arch:          linux/s390x
  Experimental:     false
 containerd:
  Version:          1.5.5-0ubuntu3~18.04.2
  GitCommit:
 runc:
  Version:          1.0.1-0ubuntu2~18.04.1
  GitCommit:
 docker-init:
  Version:          0.19.0
  GitCommit:

# docker build --help

Usage:  docker build [OPTIONS] PATH | URL | -

Build an image from a Dockerfile

Options:
      --add-host list           Add a custom host-to-IP mapping (host:ip)
      --build-arg list          Set build-time variables
      --cache-from strings      Images to consider as cache sources
      --cgroup-parent string    Optional parent cgroup for the container
      --compress                Compress the build context using gzip
      --cpu-period int          Limit the CPU CFS (Completely Fair Scheduler) period
      --cpu-quota int           Limit the CPU CFS (Completely Fair Scheduler) quota
  -c, --cpu-shares int          CPU shares (relative weight)
      --cpuset-cpus string      CPUs in which to allow execution (0-3, 0,1)
      --cpuset-mems string      MEMs in which to allow execution (0-3, 0,1)
      --disable-content-trust   Skip image verification (default true)
  -f, --file string             Name of the Dockerfile (Default is 'PATH/Dockerfile')
      --force-rm                Always remove intermediate containers
      --iidfile string          Write the image ID to the file
      --isolation string        Container isolation technology
      --label list              Set metadata for an image
  -m, --memory bytes            Memory limit
      --memory-swap bytes       Swap limit equal to memory plus swap: '-1' to enable unlimited swap
      --network string          Set the networking mode for the RUN instructions during build
                                (default "default")
      --no-cache                Do not use cache when building the image
      --pull                    Always attempt to pull a newer version of the image
  -q, --quiet                   Suppress the build output and print image ID on success
      --rm                      Remove intermediate containers after a successful build
                                (default true)
      --security-opt strings    Security options
      --shm-size bytes          Size of /dev/shm
  -t, --tag list                Name and optionally a tag in the 'name:tag' format
      --target string           Set the target build stage to build.
      --ulimit ulimit           Ulimit options (default [])
```
For s390x, what I'm thinking is if user has a s390x box, he/she can build the go-build image first, a separate PR created here: https://github.com/projectcalico/go-build/pull/379, then build the calico images with the built go-build on same box. Or, maybe we can also push the s390x go-build image to docker hub. Do you know how to push the s390x go-build image to docker hub via CI system? Do we have s390x box in CI?

[the remark this answers is not in the document]
Comment (caseydavenport) [reply apiserver/Makefile:115]: @huoqifeng did we end up getting the s390x images available in dockerhub?

[the remark this answers is not in the document]
Comment (huoqifeng) [reply apiserver/Makefile:115]: > @huoqifeng did we end up getting the s390x images available in dockerhub?

@caseydavenport right, we can keep align with other ARCHs after merge https://github.com/projectcalico/go-build/pull/379 and get a s390x `go-build` image.

**Comment by huoqifeng** [comment]
@caseydavenport I'll drop the `EXCLUDEARCH?=` for s390x in this PR, it could only be merged after https://github.com/projectcalico/calico/pull/5793 and https://github.com/projectcalico/go-build/pull/379 get merged.

**Comment by caseydavenport** [comment]
@huoqifeng maybe we can just implement this change as part of https://github.com/projectcalico/calico/pull/5793? They seem pretty tightly coupled.

**Pull request closed without merging**